amqphandler/amqphandler_test.go

```diff
@@ -644,3 +857,125 @@ func TestReceiveMessages(t *testing.T) {
 		}
 	}
 }
+
+func TestDerviceDescovery(t *testing.T) {
+	cryptoContext := &testCryptoContext{}
+
+	time.Sleep(time.Second)
+
+	amqpHandler, err := amqphandler.New()
+	if err != nil {
+		t.Fatalf("Can't create amqp: %v", err)
+	}
+	defer amqpHandler.Close()
+
+	if err := amqpHandler.Connect(cryptoContext, "http://:8010/", systemID, true); err != nil {
+		t.Errorf("Can't establish connection: %v", err)
+	}
+}
+
+/***********************************************************************************************************************
+ * Interfaces
+ **********************************************************************************************************************/
+
+func (context *testCryptoContext) GetTLSConfig() (config *tls.Config, err error) {
+	return nil, err
+}
+
+func (context *testCryptoContext) DecryptMetadata(input []byte) (output []byte, err error) {
+	switch string(input) {
+	case "boardconfig":
+		output, err = json.Marshal(context.decodedData.BoardConfig)
+
+	case "services":
+		output, err = json.Marshal(context.decodedData.Services)
+
+	case "layers":
+		output, err = json.Marshal(context.decodedData.Layers)
+
+	case "instances":
+		output, err = json.Marshal(context.decodedData.Instances)
+
+	case "components":
+		output, err = json.Marshal(context.decodedData.Components)
+
+	case "fotaschedule":
+		output, err = json.Marshal(context.decodedData.FOTASchedule)
+
+	case "sotashedule":
+		output, err = json.Marshal(context.decodedData.SOTASchedule)
+
+	case "error":
+
+	default:
+		err = aoserrors.New("Incorrect desired data")
+	}
+
+	return output, aoserrors.Wrap(err)
+}
+
+/***********************************************************************************************************************
+ * Private
+ **********************************************************************************************************************/
+
+func waitMessage(messageChannel <-chan amqphandler.Message, timeout time.Duration) (amqphandler.Message, error) {
+	select {
+	case <-time.After(timeout):
+		return amqphandler.Message{}, errTimeout
+
+	case message := <-messageChannel:
+		return message, nil
+	}
+}
+
+func serviceDiscovery(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.Error(w, "404 not found.", http.StatusNotFound)
```

Comment:
no need to put code into error message. Just 'path not found`.

Reply:
removed